speech/api/speech_streaming.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/python
+
+import contextlib
+import threading
+
+from gcloud.credentials import get_credentials
+
+from google.cloud.speech.v1.cloud_speech_pb2 import *  # noqa
+from google.rpc import code_pb2
+
+from grpc.beta import implementations
+
+import pyaudio
+
+# Audio recording parameters
+RATE = 16000
+CHANNELS = 1
+CHUNK = RATE // 10  # 100ms
+
+# Keep the request alive for this many seconds
+DEADLINE_SECS = 8 * 60 * 60
+SPEECH_SCOPE = 'https://www.googleapis.com/auth/cloud-platform'
+
+
+def _make_channel(host, port):
+    """Creates an SSL channel with auth credentials from the environment."""
+    # In order to make an https call, use an ssl channel with defaults
+    ssl_channel = implementations.ssl_channel_credentials(None, None, None)
+
+    # Grab application default credentials from the environment
+    creds = get_credentials().create_scoped([SPEECH_SCOPE])
+    # Add a plugin to inject the creds into the header
+    auth_header = (
+            'Authorization',
+            'Bearer ' + creds.get_access_token().access_token)
+    auth_plugin = implementations.metadata_call_credentials(
+            lambda _, cb: cb([auth_header], None),
+            name='google_creds')
+
+    # compose the two together for both ssl and google auth
+    composite_channel = implementations.composite_channel_credentials(
+            ssl_channel, auth_plugin)
+
+    return implementations.secure_channel(host, port, composite_channel)
+
+
+@contextlib.contextmanager
+def _record_audio(channels, rate, chunk):
+    """Opens a recording stream in a context manager."""
+    p = pyaudio.PyAudio()
```

Comment:
But that's what [pyaudio](https://people.csail.mit.edu/hubert/pyaudio/#examples) does!

...will fix.